internal/metrics/handler.go

```diff
@@ -0,0 +1,57 @@
+// SPDX-FileCopyrightText: 2025 The Pion community <https://pion.ly>
+// SPDX-License-Identifier: MIT
+
+// Package metrics provides service to expose Ion's metric.
+package metrics
+
+import (
+	"net/http"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+)
+
+type Auth struct {
+	Mode       string // "none"|"basic"|"bearer"
+	User, Pass string
+	Token      string
+}
+
+// Handler returns a http handler wrapped with authentication.
+func (s *PromService) Handler(auth Auth) http.Handler {
+	h := promhttp.HandlerFor(s.reg, promhttp.HandlerOpts{})
+	switch auth.Mode {
+	case "basic":
+		return basicAuth(h, auth.User, auth.Pass)
+	case "bearer":
+		return bearerAuth(h, auth.Token)
+	default:
+		return h
+	}
+}
```

Comment:
What about using Pion's options patterns so we can extended it more in the future, I'm actually going to add two options after your PR.

what do you think?

```suggestion
type authMode int

const (
	modeNone authMode = iota
	modeBasic
	modeBearer
)

type authConfig struct {
	mode       authMode
	user, pass string
	token      string
}

type Option func(*authConfig)

func WithBasicAuth(user, pass string) Option {
	return func(c *authConfig) {
		c.mode = modeBasic
		c.user = user
		c.pass = pass
	}
}

func WithBearerToken(token string) Option {
	return func(c *authConfig) {
		c.mode = modeBearer
		c.token = token
	}
}

func (s *PromService) Handler(opts ...Option) http.Handler {
	cfg := authConfig{mode: modeNone}
	for _, opt := range opts {
		opt(&cfg)
	}

	h := promhttp.HandlerFor(s.reg, promhttp.HandlerOpts{})

	switch cfg.mode {
	case modeBasic:
		return basicAuth(h, cfg.user, cfg.pass)
	case modeBearer:
		return bearerAuth(h, cfg.token)
	default:
		return h
	}
}
```

Reply:
This looks like a great pattern!